test/test_predict_command.py

```diff
@@ -34,25 +35,119 @@ def setup_module():
 def test_csv():
     args = ["--allele-column", "Allele", "--peptide-column", "Peptide"]
     deletes = []
+    result = None
     try:
         with tempfile.NamedTemporaryFile(delete=False, suffix=".csv") as fd:
             fd.write(TEST_CSV.encode())
             deletes.append(fd.name)
-        fd_out = tempfile.NamedTemporaryFile(delete=False, suffix=".csv")
-        deletes.append(fd_out.name)
-        full_args = [fd.name] + args + ["--out", fd_out.name]
-        print("Running with args: %s" % full_args)
-        predict_command.run(full_args)
-        result = pandas.read_csv(fd_out.name)
-        print(result)
-        assert not result.isnull().any().any()
+            fd.close()  # Explicitly close
+            
+        with tempfile.NamedTemporaryFile(delete=False, suffix=".csv") as fd_out:
+            deletes.append(fd_out.name)
+            full_args = [fd.name] + args + ["--out", fd_out.name]
+            print("Running with args: %s" % full_args)
+            predict_command.run(full_args)
+            fd_out.close()  # Explicitly close
+            result = pandas.read_csv(fd_out.name)
+            print(result)
+            assert not result.isnull().any().any()
     finally:
+        # Make sure we've closed the files and pandas has released them
+        import gc
+        gc.collect()  # Force garbage collection
+        
+        # Add a small delay to ensure files are released
+        import time
+        time.sleep(0.1)
+        
         for delete in deletes:
-            os.unlink(delete)
+            try:
+                os.unlink(delete)
+            except Exception as e:
+                print(f"Warning: Could not delete {delete}: {e}")
 
     assert result.shape == (3, 8)
 
 
+def test_tensorflow_vs_pytorch_backends():
+    """Test that tensorflow and pytorch backends produce matching results."""
+    args = [
+        "--alleles", "HLA-A0201",
```

Comment:
For the final version of this I would feel more confident if we have a test that runs a presentation model on at least a few thousand sequences and checks that the binding affinity, processing score, and presentation score all match between torch and tensorflow

Reply:
Sounds great. Do you have a dataset you'd like me to use as a test dataset?

Reply:
We can use random peptides here, just make sure they span a few different lengths. This function can help: https://github.com/openvax/mhcflurry/blob/master/mhcflurry/common.py#L171

Reply:
I've replaced with random peptides in this test and made a note to do `binding affinity, processing score, and presentation score`.